### PR TITLE
refactor: drop bluebird dependency

### PR DESCRIPTION
**Pull request by chmelevskij**
Drops bluebird dependency in favour of native apis.

Needs testing in presets, cli and osd tabs.

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by haslinghuis** [comment]
@chmelevskij please fix test units

File: src/js/tabs/osd.js [hunk 1]
```diff
@@ -253,10 +253,18 @@ FONT.msp = {
...
+        .reduce(
+            (p, x, i) =>
+                p.then(() => {
+                    $progress.val((i / FONT.data.characters.length) * 100);
+                    return MSP.promise(
+                        MSPCodes.MSP_OSD_CHAR_WRITE,
+                        FONT.msp.encode(i),
+                    );
+                }),
+            Promise.resolve(),
```
Comment (haslinghuis) [inline src/js/tabs/osd.js:266]: no argument?

**Comment by haslinghuis** [comment]
osd, cli tested.

preview has error:

image

File: src/js/tabs/osd.js (hunk 1 above)
Comment (haslinghuis) [reply src/js/tabs/osd.js:266]: > TBH. Could probably even use for..of here with awaiting but wanted to stay closer to previious api + prefer FP more

Would prefer Functional Programming here too. Rewriting using await is more clear (to me anyways)

**Review by haslinghuis — APPROVED** [review]

**Review by blckmn — APPROVED** [review]

**Review by blckmn — APPROVED** [review]

**Review by blckmn — APPROVED** [review]

**Pull request merged**